catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -269,11 +264,31 @@ public SolrQuery propertyIsEqualTo(String propertyName, String literal,
         }
         verifyInputData(propertyName, literal);
 
-        String mappedPropertyName = getMappedPropertyName(propertyName,
-                AttributeFormat.STRING,
-                true);
-        return new SolrQuery(
-                mappedPropertyName + ":" + QUOTE + escapeSpecialCharacters(literal) + QUOTE);
+        String searchPhrase = QUOTE + escapeSpecialCharacters(literal) + QUOTE;
+
+        if (Metacard.ANY_TEXT.equals(propertyName)) {
+            return new SolrQuery(anyTextSolrQuery(searchPhrase, true));
+        } else {
+            String mappedPropertyName = getMappedPropertyName(propertyName,
+                    AttributeFormat.STRING,
+                    true);
+            return new SolrQuery(mappedPropertyName + ":" + searchPhrase);
+        }
+    }
+
+    private String anyTextSolrQuery(String searchPhrase, boolean isCaseSensitive) {
+        String solrQuery = resolver.anyTextFields()
+                .map(resolver::getWhitespaceTokenizedField)
+                .map((whitespaceField) -> {
+                    if (isCaseSensitive) {
+                        return resolver.getCaseSensitiveField(whitespaceField);
+                    } else {
+                        return whitespaceField;
+                    }
+                })
+                .map((field) -> field + ":" + searchPhrase)
+                .collect(Collectors.joining(" "));
+        return "(" + solrQuery + ")";
```

Comment:
You could skip this last line by using the overloaded version of `joining` that takes prefix and suffix.